src/mbgl/map/map_context.cpp

```diff
@@ -216,6 +216,12 @@ void MapContext::render() {
     // Cleanup OpenGL objects that we abandoned since the last render call.
     env.performCleanup();
 
+    if (data.mode == MapMode::Still &&
+        (!callback || !style->isLoaded() || !resourceLoader->getSprite()->isLoaded())) {
```

Comment:
I think `resourceLoader->getSprite()->isLoaded()` is probably not needed. `style->isLoaded()` will only be true if all the tiles are in the `parsed` state. If the sprite is not loaded, some tile there will be `partial`.